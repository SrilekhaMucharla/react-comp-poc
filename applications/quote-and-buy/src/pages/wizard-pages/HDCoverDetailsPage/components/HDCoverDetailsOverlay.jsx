import React from 'react';
import PropTypes from 'prop-types';
import {
    HDLabelRefactor,
    HDQuoteInfoRefactor
} from 'hastings-components';
import { Row, Col } from 'react-bootstrap';
import {
    AnalyticsHDOverlayPopup as HDOverlayPopup,
    AnalyticsHDDropdownList as HDDropdownList,
} from '../../../../web-analytics';
import {
    seeFullDetailsLink, coverType, customizeQuote, coverDetailsOverlayPriceInfo
} from '../HDCoverDetails.messages';
import Tick from '../../../../assets/images/icons/tick-icon.svg';
import { pageMetadataPropTypes } from '../../../../constant/propTypes';

export default function HDCoverDetailsOverlay({
    children,
    branchName,
    onChange,
    policyStartDate,
    paymentTypeText,
    benefits,
    selectedCoverType,
    options,
    pageMetadata,
    headerBar
}) {
    const isDropdownDisabled = branchName === 'Hastings Essential';

    return (
        <HDOverlayPopup
            webAnalyticsView={{ ...pageMetadata, page_section: `${customizeQuote} - ${seeFullDetailsLink}` }}
            webAnalyticsEvent={{ event_action: `${customizeQuote} - ${seeFullDetailsLink}` }}
            id="cover-details-overlay"
            showButtons={false}
            labelText="Your policy details"
            overlayButtonIcon={(
                <HDLabelRefactor
                    Tag="a"
                    className="decorated-blue-line theme-white"
                    text={seeFullDetailsLink} />
            )}
            headerBar={headerBar}
        >
            <hr />
            <Row className="cover-details__overlay-policy-header">
                <Col>
                    <Row className="mb-2">
                        <Col xs={5}>
                            <HDLabelRefactor
                                Tag="span"
                                text="Cover level:" />
                        </Col>
                        <Col>
                            <HDLabelRefactor
                                Tag="span"
                                className="font-medium"
                                text={branchName} />
                        </Col>
                    </Row>
                    <Row className="mb-2">
                        <Col xs={5} className="align-self-center">
                            <HDLabelRefactor
                                Tag="span"
                                className="align-items-center"
                                text="Cover type:" />
                        </Col>
                        <Col>
                            <HDDropdownList
                                webAnalyticsEvent={{ event_action: `${customizeQuote} - ${coverType}` }}
                                id="cover-type-dropdown"
                                customClassName="cover-details__overlay-dropdown font-medium"
                                isDisabled={isDropdownDisabled}
                                options={options}
                                value={selectedCoverType}
                                onChange={onChange} />
                        </Col>
                    </Row>
                    <Row className="cover-details__overlay-policy-row mb-2">
                        <Col>
                            <HDQuoteInfoRefactor className="customize-quote-summary__quote-notice__quote-info text-regular">
                                <div>{coverDetailsOverlayPriceInfo}</div>
                            </HDQuoteInfoRefactor>
                        </Col>
                    </Row>
                    <Row className="mb-2">
                        <Col xs={5}>
                            <HDLabelRefactor
                                Tag="span"
                                text="Start date:" />
                        </Col>
                        <Col>
                            <HDLabelRefactor
                                Tag="span"
                                className="font-medium"
                                text={policyStartDate} />
                        </Col>
                    </Row>
                    <Row>
                        <Col xs={5}>
                            <HDLabelRefactor
                                Tag="span"
                                text="Payment terms:" />
                        </Col>
                        <Col>
                            <HDLabelRefactor
                                Tag="span"
                                className="font-medium"
                                text={paymentTypeText} />
                        </Col>
                    </Row>
                </Col>
            </Row>
            <hr />
            <Row className="mb-3">
                <Col>
                    <HDLabelRefactor
                        Tag="h5"
                        className="m-0 font-weight-bold"
                        text="What's included:" />
                </Col>
            </Row>
            <Row className="cover-details__overlay-benefits-list d-block">
                {benefits.map((benefit) => (
                    <Col>
                        <Row className="cover-details__overlay-benefits font-medium">
                            <Col>
                                <HDLabelRefactor
                                    Tag="span"
                                    text={benefit.label}
                                    icon={<img src={Tick} alt="tick" />}
                                    iconPosition="l" />
                            </Col>
                        </Row>
                        <Row className="cover-details__overlay-description">
                            <Col>
                                <HDLabelRefactor
                                    Tag="span"
                                    text={benefit.description} />
                            </Col>
                        </Row>
                    </Col>
                ))}
            </Row>
            {children}
        </HDOverlayPopup>
    );
}

HDCoverDetailsOverlay.propTypes = {
    pageMetadata: PropTypes.shape(pageMetadataPropTypes).isRequired,
    children: PropTypes.node.isRequired,
    branchName: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    policyStartDate: PropTypes.string.isRequired,
    paymentTypeText: PropTypes.string.isRequired,
    benefits: PropTypes.arrayOf(PropTypes.shape({ benefit: PropTypes.string, description: PropTypes.string })).isRequired,
    selectedCoverType: PropTypes.shape({ value: PropTypes.oneOf(['comprehensive', 'thirdParty']), label: PropTypes.string }).isRequired,
    options: PropTypes.arrayOf(PropTypes.shape({ value: PropTypes.oneOf(['comprehensive', 'thirdParty']), label: PropTypes.string })).isRequired,
    headerBar: PropTypes.shape({})
};

HDCoverDetailsOverlay.defaultProps = {
    headerBar: null
};
