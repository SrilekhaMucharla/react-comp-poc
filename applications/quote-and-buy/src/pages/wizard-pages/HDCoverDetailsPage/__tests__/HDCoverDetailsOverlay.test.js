import React from 'react';
import { shallow } from 'enzyme';
import HDCoverDetailsOverlay from '../components/HDCoverDetailsOverlay';

describe('<HDCoverDetailsOverlay />', () => {
    it('should render correctly and match the snapshot', () => {
        // given
        const component = shallow(
            <HDCoverDetailsOverlay
                branchName="HD"
                onChange={jest.fn()}
                policyStartDate="20/05/2021"
                paymentTypeText="Montlhy"
                benefits={[]}
                selectedCoverType={{ value: 'comprehensive', label: 'Comprehensive' }}
                options={[{ value: 'comprehensive', label: 'Comprehensive' }, { value: 'thirdParty', label: 'Third Party' }]}
            >
                <p>Content</p>
            </HDCoverDetailsOverlay>
        );
        // then
        expect(component).toMatchSnapshot();
    });
});
