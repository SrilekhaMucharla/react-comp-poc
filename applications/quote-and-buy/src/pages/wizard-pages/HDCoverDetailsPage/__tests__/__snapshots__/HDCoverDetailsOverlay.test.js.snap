// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDCoverDetailsOverlay /> should render correctly and match the snapshot 1`] = `
<ForwardRef
  headerBar={null}
  id="cover-details-overlay"
  labelText="Your policy details"
  onBeforeOpen={null}
  overlayButtonIcon={
    <HDLabelRefactor
      Tag="a"
      adjustImagePosition={true}
      className="decorated-blue-line theme-white"
      text="See full details"
    />
  }
  showButtons={false}
  webAnalyticsEvent={
    Object {
      "event_action": "Customize quote - See full details",
    }
  }
  webAnalyticsView={
    Object {
      "page_section": "Customize quote - See full details",
    }
  }
>
  <hr />
  <Row
    className="cover-details__overlay-policy-header"
    noGutters={false}
  >
    <Col>
      <Row
        className="mb-2"
        noGutters={false}
      >
        <Col
          xs={5}
        >
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className=""
            text="Cover level:"
          />
        </Col>
        <Col>
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className="font-medium"
            text="HD"
          />
        </Col>
      </Row>
      <Row
        className="mb-2"
        noGutters={false}
      >
        <Col
          className="align-self-center"
          xs={5}
        >
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className="align-items-center"
            text="Cover type:"
          />
        </Col>
        <Col>
          <ForwardRef
            customClassName="cover-details__overlay-dropdown font-medium"
            id="cover-type-dropdown"
            isDisabled={false}
            onBeforeOpen={null}
            onBlur={null}
            onCancel={null}
            onChange={[MockFunction]}
            onClick={null}
            onConfirm={null}
            onDelete={null}
            onEdit={null}
            onFocus={null}
            onSelect={null}
            options={
              Array [
                Object {
                  "label": "Comprehensive",
                  "value": "comprehensive",
                },
                Object {
                  "label": "Third Party",
                  "value": "thirdParty",
                },
              ]
            }
            value={
              Object {
                "label": "Comprehensive",
                "value": "comprehensive",
              }
            }
            webAnalyticsEvent={
              Object {
                "event_action": "Customize quote - Cover Type",
              }
            }
          />
        </Col>
      </Row>
      <Row
        className="cover-details__overlay-policy-row mb-2"
        noGutters={false}
      >
        <Col>
          <HDQuoteInfoRefactor
            className="customize-quote-summary__quote-notice__quote-info text-regular"
          >
            <div>
              Sorry, if your car is worth more than Â£9,000 only comprehensive cover is available.
            </div>
          </HDQuoteInfoRefactor>
        </Col>
      </Row>
      <Row
        className="mb-2"
        noGutters={false}
      >
        <Col
          xs={5}
        >
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className=""
            text="Start date:"
          />
        </Col>
        <Col>
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className="font-medium"
            text="20/05/2021"
          />
        </Col>
      </Row>
      <Row
        noGutters={false}
      >
        <Col
          xs={5}
        >
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className=""
            text="Payment terms:"
          />
        </Col>
        <Col>
          <HDLabelRefactor
            Tag="span"
            adjustImagePosition={true}
            className="font-medium"
            text="Montlhy"
          />
        </Col>
      </Row>
    </Col>
  </Row>
  <hr />
  <Row
    className="mb-3"
    noGutters={false}
  >
    <Col>
      <HDLabelRefactor
        Tag="h5"
        adjustImagePosition={true}
        className="m-0 font-weight-bold"
        text="What's included:"
      />
    </Col>
  </Row>
  <Row
    className="cover-details__overlay-benefits-list d-block"
    noGutters={false}
  />
  <p>
    Content
  </p>
</ForwardRef>
`;
